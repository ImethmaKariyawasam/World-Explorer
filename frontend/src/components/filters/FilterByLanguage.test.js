import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import FilterByLanguage from './FilterByLanguage';

describe('FilterByLanguage Component', () => {
  const mockLanguages = ['English', 'Spanish', 'French', 'German'];
  const mockOnChange = jest.fn();

  beforeEach(() => {
    mockOnChange.mockClear();
  });

  it('renders with default option and language options', () => {
    render(
      <FilterByLanguage 
        languages={mockLanguages} 
        selectedLanguage="" 
        onLanguageChange={mockOnChange} 
      />
    );

    expect(screen.getByRole('combobox')).toBeInTheDocument();
    expect(screen.getByText('Filter by Language')).toBeInTheDocument();
    mockLanguages.forEach(lang => {
      expect(screen.getByText(lang)).toBeInTheDocument();
    });
  });

  it('shows the selected language', () => {
    render(
      <FilterByLanguage 
        languages={mockLanguages} 
        selectedLanguage="French" 
        onLanguageChange={mockOnChange} 
      />
    );

    expect(screen.getByDisplayValue('French')).toBeInTheDocument();
  });

  it('calls onLanguageChange when selection changes', () => {
    render(
      <FilterByLanguage 
        languages={mockLanguages} 
        selectedLanguage="" 
        onLanguageChange={mockOnChange} 
      />
    );

    const select = screen.getByRole('combobox');
    fireEvent.change(select, { target: { value: 'Spanish' } });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledWith('Spanish');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <FilterByLanguage 
        languages={mockLanguages} 
        selectedLanguage="English" 
        onLanguageChange={mockOnChange} 
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});