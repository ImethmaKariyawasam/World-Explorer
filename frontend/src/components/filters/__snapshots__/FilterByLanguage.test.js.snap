// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByLanguage Component matches snapshot 1`] = `
<DocumentFragment>
  <select
    aria-label="Select language"
    class="form-select"
    style="width: 200px;"
  >
    <option
      value=""
    >
      Filter by Language
    </option>
    <option
      value="English"
    >
      English
    </option>
    <option
      value="Spanish"
    >
      Spanish
    </option>
    <option
      value="French"
    >
      French
    </option>
    <option
      value="German"
    >
      German
    </option>
  </select>
</DocumentFragment>
`;
